### PR TITLE
Add the ability to center the viewport to the world

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Adds a `centerViewport` method that will automatically scale and center the graph in the viewport

Also fixes an issue where the viewportDateRange was not set when the viewport was started.

### CODE REVIEW

File: src/objects/nodes.ts [hunk 1]
```diff
@@ -78,8 +109,8 @@ async function createNode(node: RunGraphNode): Promise<NodeSprites> {
     label,
   })
 
-  viewport.addChild(graphics)
-  viewport.addChild(label)
+  container!.addChild(graphics)
+  container!.addChild(label)
```
Comment (znicholasbrown) [inline src/objects/nodes.ts:113]: why the non-null assertion? is this coming from pixi and not well-typed?

**Review by znicholasbrown — APPROVED** [review]
one question otherwise lgtm

also where'd my deploy preview go

File: src/objects/nodes.ts (hunk 1 above)
Comment (pleek91) [reply src/objects/nodes.ts:113]: oh I forgot to come back to this. I didn't care to fix it until I knew it was working. Will update

Comment (pleek91) [reply src/objects/nodes.ts]: Fixed!

**Comment by pleek91** [comment]
Deploy preview probably didn't show up until this was updated to PR into main, this was chained off another branch originally.

**Pull request merged**